ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -237,12 +238,11 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
       "A transaction should be visible to a non-submitting stakeholder but its command identifier should be empty"
     ) { context =>
       for {
-        ledger <- context.participant()
-        Vector(submitter, listener) <- ledger.allocateParties(2)
-        (id, _) <- ledger.createAndGetTransactionId(
-          submitter,
-          AgreementFactory(listener, submitter))
-        tree <- ledger.transactionTreeById(id, listener)
+        Vector(alpha, beta) <- context.participants(2)
+        submitter <- alpha.allocateParty()
+        listener <- beta.allocateParty()
+        (id, _) <- alpha.createAndGetTransactionId(submitter, AgreementFactory(listener, submitter))
+        tree <- beta.transactionTreeById(id, listener)
```

Comment:
Potential flakiness between `alpha` and `beta`.

Reply:
Since this tests tries do "prove non-existence" of something, it looks like it's less of a case of potential flakiness and more of a source of false positives, like in the case of https://github.com/digital-asset/daml/pull/2900#discussion_r324545682. Makes sense?

Reply:
In this case it's flakiness, because we're not trying to prove that the transaction is not visible, but rather that the `commandId` is not set.

Reply:
Of course, thanks, I'll wrap the call to `transactionTreeById` with `eventually`.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2900/commits/cb3e0358d915d8511f21e83057e70a1616c6a3a9